src/base/bittorrent/torrenthandle.h

```diff
@@ -282,6 +282,7 @@ namespace BitTorrent
         bool isErrored() const;
         bool isSequentialDownload() const;
         bool hasFirstLastPiecePriority() const;
+        bool isPreviewable() const;
```

Comment:
Does it make sense to have this function reside in `TorrentInfo` class instead?

Reply:
On the other hand, isn't it too trivial to separate it into a separate method?

Reply:
Ok, I move it to TorrentInfo and change name to hasPreviewableFiles, will it be ok?

Reply:
> On the other hand, isn't it too trivial to separate it into a separate method?

What do you mean, create private method in `TransferListWidget` class?

Reply:
>create private method in TransferListWidget class?

Are you C#/Java programmer?
Why should it be TransferListWidget class method? It has nothing to do with this class. So just create helper function (e.g. in anonymous namespace in transferlistwidget.cpp).

Reply:
Understand, but noo, that is ugly :/, don't like this declaration

Reply:
Can make it private in `TorrentInfo` class, if you don't want it to be part of an api

Reply:
>that is ugly :/, don't like this declaration

Every thing should be in its place. What does it have to do with beautiful or ugly?
>Can make it private in TorrentInfo class

???

Reply:
Is awareness of file types is part of "torrent file" logic? No. So why it should be in TorrentInfo class (that represents "torrent file")?

Reply:
From my view is better place `TorrentHandle` class, it is regular question if torrent handle is previewable ( it could be part of libtorrent_handle api too ) , `TorrentInfo` class is correct too, both are ok, regular business logic.

Reply:
> it is regular question if torrent handle is previewable

Incorrect wording as well as incorrect meaning. Regular question is "does torrent contain previewable files?". This knowledge isn't required by TorrentHandle and TorrentInfo for doing its job completely. So it's out of its "business logic".
Following your judgment, any analytical action on some model is part of its business logic. For example, `TorrentInfo::hasBigFiles()`, `TorrentInfo::hasAudioFiles()`, `TorrentInfo::hasFilesWithLongNames()` etc.

Reply:
So it can be part of library class, `Utils::Misc::hasPreviewableFiles()` is good candidate.

Reply:
My previous comment:
>Are you C#/Java programmer?
Why should it be TransferListWidget class method? It has nothing to do with this class. So just create helper function (e.g. in anonymous namespace in transferlistwidget.cpp).

Reply:
Pls propose any other solution than function outside the class.

Reply:
What's the big deal?

Reply:
Disarranged look, makes code more unreadable, but this pattern is used across whole qbt, so doesn't matter.

Reply:
Why disarranged?

Reply:
I'm used to have almost all functionality part of the class, I'm preferring to declare class method against local function, imagine how looks class with 3 methods vs class with 2 methods and local function in anonym. namespace, because of this exists private access modifier.
I'm looking through qt source files and this patter is rarely used.

Reply:
>I'm preferring to declare class method against local function

It's just your baseless preference...

If a function does not deal with private members of a class, it is not a class method in its essence. (However, this statement does not mean that if a function should have access to private members of the class, it must be a method of the class itself, it often makes sense to define it as a friend function).

If all my arguments sound unconvincing to you, then you can search the Internet for the opinions of more famous people.

Either way, you have to take it as an "unwritten" Coding Style rule of qBittorrent project.